### PR TITLE
Setup field match executor shared state.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/features/fieldmatchfeature.cpp [hunk 1]
```diff
@@ -18,35 +18,50 @@ using CollectionType = FieldInfo::CollectionType;
...
+};
+
+FieldMatchExecutorSharedState::FieldMatchExecutorSharedState(const IQueryEnvironment& query_env,
+                                                             const FieldInfo& field,
+                                                             const fieldmatch::Params& params)
+    : Anything(),
+      _splitter_env(query_env, field.id()),
+      _cmp_shared_state(vespalib::make_string("fieldMatch(%s)", field.name().c_str()), _splitter_env, field, params)
+{
+    // empty
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/fieldmatchfeature.cpp:41]: This is not needed, consider removing

**Review by geirst — APPROVED** [review]
👍

File: searchlib/src/vespa/searchlib/features/fieldmatchfeature.cpp (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/features/fieldmatchfeature.cpp]: Removed.

**Review by geirst — APPROVED** [review]
👍

**Comment by jonmv** [comment]
Broke some system tests.

**Pull request merged**